### PR TITLE
REF: rearrange test_to_latex.py

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Rearranged tests in ``test_to_latex.py`` to the corresponding classes.
Sorry for the messy diffs.
All I did (except to small commits cd92db9 and 2b66e2f) was rearrangement of the tests.

### CODE REVIEW

**Comment by jreback** [comment]
can you just confirm that before & after test counts are the same for latex.

**Comment by ivanovmg** [comment]
"Added" tests:
```
$ git diff e260f58^..8815c7f | grep "^+.*def test_" | wc -l
44
```

"Deleted" tests:
```
$ git diff e260f58^..8815c7f | grep "^-.*def test_" | wc -l
43
```

One extra test was added: ``test_to_latex_column_format_just_works`` as a result of splitting ``test_to_latex_column_format``.
Originally ``test_to_latex_column_format`` basically comprised two tests: one to tests that column format just works without failing for dataframe ``float_frame`` (fixture) and the second one to test result vs expected for small dataframe. I extracted the test ``_just_work`` for the ``float_frame``.

File: pandas/tests/io/formats/test_to_latex.py [hunk 1]
```diff
@@ -152,10 +100,11 @@ def test_to_latex_bad_column_format(self, bad_column_format):
         with pytest.raises(ValueError, match=msg):
             df.to_latex(column_format=bad_column_format)
 
-    def test_to_latex_column_format(self, float_frame):
+    def test_to_latex_column_format_just_works(self, float_frame):
         # GH Bug #9402
         float_frame.to_latex(column_format="lcr")
 
+    def test_to_latex_column_format(self):
         df = DataFrame({"a": [1, 2], "b": ["b1", "b2"]})
```
Comment (ivanovmg) [inline pandas/tests/io/formats/test_to_latex.py:108]: This is where I extracted tests ``_just_works``.

**Comment by jreback** [comment]
thanks @ivanovmg

**Pull request merged**